aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -310,7 +310,11 @@ std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool kee
               "expected scalar type '", ScalarType::Long, "' but got '",
               indices.scalar_type(), "' for indices output");
   dim = maybe_wrap_dim(dim, self.dim());
-  if (_dimreduce_return_trivial_no_ident(values, self, dim, keepdim, "mode")) {
+  if (self.numel() == 0) {
+    zero_numel_tensor_resize(values, indices, self, dim, keepdim, "mode()");
+    return std::tie(values, indices);
```

Comment:
why is it std::tie here and std::forward_as_tuple in the next conditional?

Reply:
Its not part of this PR so I did not touch (which would involve making changes for other functions too).
Do you want me to include those changes in this PR too?